nwbinspector/checks/ecephys.py

```diff
@@ -50,3 +50,17 @@ def check_electrical_series_dims(electrical_series: ElectricalSeries):
 def check_electrical_series_reference_electrodes_table(electrical_series: ElectricalSeries):
     if electrical_series.electrodes.table.name != "electrodes":
         return InspectorMessage(message="electrodes does not  reference an electrodes table.")
+
+
+@register_check(importance=Importance.BEST_PRACTICE_VIOLATION, neurodata_type=Units)
+def check_spike_times_not_in_unobserved_interval(units_table: Units, nelems: int = 200):
+    """Check if a Units table has spike times that occur outside of observed intervals."""
+    if units_table.obs_intervals:
```

Comment:
I prefer early stopping. Less indented code and easier to read as you don't need to check far down the line for an `else` clause.

```suggestion
    if not units_table.obs_intervals:
        return
```